doc/Makefile

```diff
@@ -19,25 +19,68 @@ default: container-build
 # should not be used with `make -f`.
 ##
 
+CONTAINER_VERSION=1.1.1
+CONTAINER_NAME=openassetio-doc-build
+
+##
+## Container settings
+##
+
+# The following variables are considered when using Docker
+# to build the documentation:
+#
+#  GITHUB_ORG: The organisation to pull containers from (and publish
+#              to). If unspecified, then `thefoundryvisionmongers` will
+#              be used. Note: org names must be lowercase.
+#
+# When publishing, the following additional vars MUST be set:
+#
+#  GITHUB_PAT:      A GitHub Personal Access Token with the following
+#                   permissions: write:packages.
+#  GITHUB_USERNAME: The GitHub username whose token is being used.
+#
+# eg:
+#  GITHUB_USER=<username> GITHUB_PAT=<token> make publish-docker
+#
+ifndef GITHUB_ORG
+GITHUB_ORG = thefoundryvisionmongers
+endif
+
+# Docker images are published to the GitHub container registry
+CONTAINER_PATH = ghcr.io/$(GITHUB_ORG)/$(CONTAINER_NAME)
+CONTAINER_TAG = $(CONTAINER_PATH):$(CONTAINER_VERSION)
 
 ##
 ## PHONY TARGETS
 ##
 
-.PHONY: clean clean-html clean-tooling clean-docker container-build deploy docker-image html todo tooling
+.PHONY: clean clean-html clean-tooling clean-docker container-build deploy docker-image html \
+        publish-docker tag-docker todo tooling
 
 #
 # Container build - Runs this Makefile via docker
 #
 
-container-build: docker-image
-	docker run --rm -v `pwd`/../:/src openassetio-doc-build bash -c 'make -C /src/doc html'
+container-build:
+	docker run --rm -v `pwd`/../:/src $(CONTAINER_TAG) bash -c 'make -C /src/doc html'
+
+# Targets for building/tagging/publishing the container
 
 docker-image: Dockerfile
-	docker build . -t openassetio-doc-build
+	docker build . -t $(CONTAINER_NAME)
 
 clean-docker:
-	docker image rm openassetio-doc-build --force
+	docker image rm $(CONTAINER_NAME) --force
+
+tag-docker: docker-image
+	docker tag $(CONTAINER_NAME) $(CONTAINER_TAG)
+
+publish-docker: tag-docker
+	echo $(GITHUB_PAT) | docker login ghcr.io --username $(GITHUB_USERNAME) --password-stdin
+	docker push $(CONTAINER_TAG)
+	# Update the 'latest' tag
+	docker tag $(CONTAINER_TAG) $(CONTAINER_PATH):latest
```

Comment:
[question] Do we need to care about the `latest` tag?

Reply:
Yeah, otherwise you have to specify an explicit version, which means you can't do:
```
docker run -v `pwd`/../:/src ghcr.io/thefoundryvisionmongers/openassetio-doc-build bash -c 'make -C /src/doc html'
```

Reply:
As in, if you want to do it manually? (since `make container-build` uses `CONTAINER_TAG`)

Reply:
Yeah (we list the above in `README.md` and it saves the maintenance overhead.